### PR TITLE
single-file-cli 2.0.14 (new cask)

### PR DESCRIPTION
**Pull request by AngelaDMerkel**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [ ] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [x] `brew audit --cask --new <cask>` worked successfully.
- [x] `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --cask <cask>` worked successfully.
- [x] `brew uninstall --cask <cask>` worked successfully.

---

Please forgive me. I never use github for anything besides my shit tier personal projects and don't have much of an idea what I am doing. Everything seems to work and all the tests produce no errors, so in theory this should be good to go. As it pertains to the correct repo I have no idea.

### CODE REVIEW

**Comment by bevanjkay** [comment]
It looks this is an open-source cli-only binary, so you try submitting it as a formula to [`homebrew-core`](https://github.com/homebrew/homebrew-core) first.

**Comment by AngelaDMerkel** [comment]
> It looks this is an open-source cli-only binary, so you try submitting it as a formula to [`homebrew-core`](https://github.com/homebrew/homebrew-core) first.

As far as I can tell from the documentation, it's required that the user can build from source. `single-file-cli` requires [Deno](https://deno.com/) and so I assumed wasn't compatible with the requirements for homebrew-core. Can `hombrew-core` accept only precompiled binaries?

File: Casks/s/single-file.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+cask "single-file" do
+  arch arm: "aarch64-apple-darwin", intel: "x86_64-apple-darwin"
+  version "2.0.14"
```
Comment (daeho-ro) [inline Casks/s/single-file.rb:3]: Insert new line before version, or run `brew style --fix single-file`.

**Comment by SMillerDev** [comment]
> single-file-cli requires [Deno](https://deno.com/) and so I assumed wasn't compatible with the requirements for homebrew-core.

Deno is open source and can be used to build formula binaries.

**Comment by AngelaDMerkel** [comment]
I am looking at the repository, and there is no real build step. `deno` is used, but the method is just to reference the `.js` files rather than compile anything. I still feel as though `cask` is the better option for this, although I could well be mistaken.

**Comment by krehel** [comment]
Closing as agreed this should be attempted as a Formula first.

**Pull request closed without merging**